system/src/system_update.cpp

```diff
@@ -99,6 +109,32 @@ void system_flag_changed(system_flag_t flag, uint8_t oldValue, uint8_t newValue)
         }
     }
 #endif // HAL_PLATFORM_POWER_MANAGEMENT_OPTIONAL
+    if (flag == SYSTEM_FLAG_OTA_UPDATE_ENABLED)
+    {
+    	// publish the firmware enabled event
+    	spark_send_event(UPDATES_ENABLED_EVENT, flag_to_string(newValue), 60, PUBLISH_EVENT_FLAG_PRIVATE, nullptr);
+    }
+    if (flag == SYSTEM_FLAG_OTA_UPDATE_FORCED)
+    {
+    	spark_send_event(UPDATES_FORCED_EVENT, flag_to_string(newValue), 60, PUBLISH_EVENT_FLAG_PRIVATE, nullptr);
```

Comment:
It looks like whenever the system receives a `particle/device/updates/forced` event it will publish exactly the same event back: https://github.com/particle-iot/device-os/pull/1732/files#diff-0e9718520afac28cc54b8a2726ef3effR382. Is that intended?

Reply:
Well spotted!  That's correct. The intent is to provide acknowledgement to the cloud that the device has received and processed the event. It helps avoid a race condition where we might try sending firmware before the event has been received.